src/plugins/application_config/server/routes/index.test.ts

```diff
@@ -0,0 +1,353 @@
+
+      expect(response.ok).toBeCalledWith({
+        body: {
+          newValue: ENTITY_NEW_VALUE,
+        },
+      });
+
+      expect(logger.error).not.toBeCalled();
+    });
+
+    it('return error response when client fails', async () => {
+      const error = new Error(ERROR_MESSAGE);
+
+      const client = {
+        updateEntityConfig: jest.fn().mockImplementation(() => {
+          throw error;
+        }),
+      };
+
+      const request = {
+        params: {
+          entity: ENTITY_NAME,
+        },
+        body: {
+          newValue: ENTITY_NEW_VALUE,
+        },
+      };
+
+      const response = {
+        customError: jest.fn().mockReturnValue(ERROR_RESPONSE),
+      };
+
+      const logger = loggerMock.create();
+
+      const returnedResponse = await handleUpdateEntityConfig(client, request, response, logger);
+
+      expect(returnedResponse).toBe(ERROR_RESPONSE);
+
+      expect(client.updateEntityConfig).toBeCalledTimes(1);
+
+      expect(response.customError).toBeCalledWith({
+        body: error,
+        statusCode: 500,
+      });
+
+      expect(logger.error).toBeCalledWith(error);
+    });
+  });
+
+  describe('handleDeleteEntityConfig', () => {
+    it('returns successful response when client succeeds', async () => {
+      const client = {
+        deleteEntityConfig: jest.fn().mockReturnValue(ENTITY_NAME),
+      };
+
+      const okResponse = {
+        statusCode: 200,
+      };
+
+      const request = {
+        params: {
+          entity: ENTITY_NAME,
+        },
+      };
+
+      const response = {
+        ok: jest.fn().mockReturnValue(okResponse),
+      };
+
+      const logger = loggerMock.create();
+
+      const returnedResponse = await handleDeleteEntityConfig(client, request, response, logger);
+
+      expect(returnedResponse).toBe(okResponse);
+
+      expect(client.deleteEntityConfig).toBeCalledTimes(1);
+
+      expect(response.ok).toBeCalledWith({
+        body: {
+          deletedEntity: ENTITY_NAME,
+        },
+      });
+
+      expect(logger.error).not.toBeCalled();
+    });
+
+    it('return error response when client fails', async () => {
+      const error = new Error(ERROR_MESSAGE);
+
+      const client = {
+        deleteEntityConfig: jest.fn().mockImplementation(() => {
+          throw error;
+        }),
+      };
+
+      const request = {
+        params: {
+          entity: ENTITY_NAME,
+        },
+      };
+
+      const response = {
+        customError: jest.fn().mockReturnValue(ERROR_RESPONSE),
+      };
+
+      const logger = loggerMock.create();
+
+      const returnedResponse = await handleDeleteEntityConfig(client, request, response, logger);
+
+      expect(returnedResponse).toBe(ERROR_RESPONSE);
+
+      expect(client.deleteEntityConfig).toBeCalledTimes(1);
+
+      expect(response.customError).toBeCalledWith({
+        body: error,
+        statusCode: 500,
+      });
+
+      expect(logger.error).toBeCalledWith(error);
+    });
+  });
+});
```

Comment:
can we add UT for custom configure client registered via registerConfigurationClient?

Reply:
The current test file [src/plugins/application_config/server/routes/index.test.ts](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5855/files#diff-5059f92734e8006e1bdd9dd51c074825d427ba7b9c12e505f5a2267905f06d3d) is about the `routers`.

The function `registerConfigurationClient` exists in the plugin class itself. Added a new test file `plugin.test.ts` to cover the `registerConfigurationClient` logic to verify error when double registering.